imports/plugins/core/layout/client/components/coreLayout.js

```diff
@@ -5,43 +5,61 @@ import { getComponent, registerComponent } from "@reactioncommerce/reaction-comp
 import Blaze from "meteor/gadicc:blaze-react-component";
 import { Template } from "meteor/templating";
 
-const CoreLayout = ({ actionViewIsOpen, structure }) => {
-  const { layoutHeader, layoutFooter, template } = structure || {};
-
-  const pageClassName = classnames({
-    "page": true,
-    "show-settings": actionViewIsOpen
-  });
-
-  const headerComponent = layoutHeader && getComponent(layoutHeader);
-  const footerComponent = layoutFooter && getComponent(layoutFooter);
-
-  let mainNode = null;
-  try {
-    const mainComponent = getComponent(template);
-    mainNode = React.createElement(mainComponent, {});
-  } catch (error) {
-    //  Probe for Blaze template (legacy)
-    if (Template[template]) {
-      mainNode = <Blaze template={template} />;
+class CoreLayout extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const { structure } = this.props;
+    const { layoutHeader, layoutFooter } = structure || {};
+
+    const headerComponent = layoutHeader && getComponent(layoutHeader);
+    const footerComponent = layoutFooter && getComponent(layoutFooter);
+
+    if (headerComponent) {
+      this.headerComponent = React.createElement(headerComponent, {});
+    }
+
+    if (footerComponent) {
+      this.footerComponent = React.createElement(footerComponent, {});
     }
   }
 
-  return (
-    <div className={pageClassName} id="reactionAppContainer">
+  render() {
+    const { actionViewIsOpen, structure } = this.props;
```

Comment:
I feel like you could have the code from lines 28 - 46 live inside a lifecycle method (componentWillMount, componentWillUpdate) and only have the return inside the `render` method.

Reply:
I thought that might be out of scope for this bug, but yes maybe it makes sense to move them now. 
Thanks :)

Reply:
@jshimko approved this so we're merging it, if that wasn't expected and @Akarshit @nnnnat  if this is worth doing please add the updates as a new PR.